hw/arm/guest-socket.c

```diff
@@ -95,25 +178,34 @@ int32_t qc_handle_bind(CPUState *cpu, int32_t sckt, struct sockaddr *g_addr,
     if (addrlen > sizeof(addr)) {
         guest_svcs_errno = ENOMEM;
     } else {
-        cpu_memory_rw_debug(cpu, (target_ulong) g_addr, (uint8_t*) &addr,
-                            sizeof(addr), 0);
+        cpu_memory_rw_debug(cpu, (target_ulong) g_addr, (uint8_t*) &ADDR,
+                            sizeof(ADDR), 0);
 
+#ifdef NETWORK_HACK
+        addr = convert_sockaddr_to(ADDR);
+#endif
```

Comment:
Similar to the change in `accept`:

```suggestion
#ifndef __APPLE__
        cpu_memory_rw_debug(cpu, (target_ulong) g_addr, (uint8_t*) &darwin_addr,
                            sizeof(darwin_addr), 0);
        addr = convert_sockaddr_to(darwin_addr);
#else
        cpu_memory_rw_debug(cpu, (target_ulong) g_addr, (uint8_t*) &addr,
                            sizeof(addr), 0);
#endif
```